### ISSUE TITLE
restartAfterPowerFailure breaks activation script

### ISSUE THREAD

**Issue by gnammix** [LnL7/nix-darwin#1236]
Hello, seems like adding `power.restartAfterPowerFailure=true` on systems that do not support this option (i.e.: my M1 pro notebook) cause the activation script to stop (e.g.: not running home-manager activation).
This is silent, no error is reported when running darwin-rebuild.

```
sudo systemsetup -setRestartPowerFailure 'on'
Restart After Power Failure: Not supported on this machine.
```

**Comment by Samasaur1** [comment]
I don't think this is something we can check against, because we don't know the Mac model that the configuration is being activated on. Additionally, I don't think we can just ignore errors with this command, because we *want* to catch other, real errors

**Comment by gnammix** [comment]
The issue I see is that the failure during activation is not reported, everything seems "normal" from user standpoint but script execution stops. If i didn't knew other part of configuration were not being applied (home-manager) I would not have noticed it.
I don't totally agree that we cannot check against it, `systemsetup` invocation in the script could be if gated or at least we could avoid stopping completely and silently in case of failures; anyway I'm really not a nix expert so cannot elaborate on that.
Just documenting it in the options description could be also a viable solution.

**Comment by Samasaur1** [comment]
I'm all for checking for this error in theory; I'm just not sure how to differentiate this (a configuration error) from any other error that systemsetup throws. I do agree that this should be mentioned in the option documentation — that should be a pretty simple PR if you'd like to tackle it. I also agree that errors during activation should be made very clear, though that's a more involved change

**Comment by Enzime** [comment]
Fixed by #1241
